src/Repository.php

```diff
@@ -20,4 +20,15 @@ public function __construct(MapperInterface $mapper)
     {
         $this->mapper = $mapper;
     }
+
+
+    /**
+     * Saves a collection of objects.
+     *
+     * @param Collection $collection
+     */
+    public function saveAll(Collection $collection)
+    {
+        $this->saveAll($collection);
```

Comment:
This will create an infinite loop.

Did you mean `$this->mapper->saveAll($collection)`?

If so, you'll need to update the `Mapper` repository to support this first.